service-registry/src/main/java/io/servicecomb/serviceregistry/client/http/RestUtils.java

```diff
@@ -126,16 +135,43 @@ public static RequestContext createRequestContext(HttpMethod method, IpPort ipPo
     return requestContext;
   }
 
+  public static SignRequest createSignRequest(RequestContext requestContext, String url, Map<String, String> headers) {
+    SignRequest signReq = new SignRequest();
+    StringBuilder endpoint = new StringBuilder("https://" + requestContext.getIpPort().getHostOrIp());
+    endpoint.append(":" + requestContext.getIpPort().getPort());
+    endpoint.append(url);
+    try {
+      signReq.setEndpoint(new URI(endpoint.toString()));
+    } catch (URISyntaxException e) {
+      LOGGER.error("set uri failed, uri is {}, message: {}", endpoint.toString(), e.getMessage());
+      return null;
+    }
+    signReq.setContent((requestContext.getParams().getBody() != null && requestContext.getParams().getBody().length > 0)
+        ? new ByteArrayInputStream(requestContext.getParams().getBody()) : null);
+    signReq.setHeaders(headers);
+    signReq.setHttpMethod(requestContext.getMethod().toString());
+    signReq.setQueryParams(requestContext.getParams().getQueryParamsMap());
+    return signReq;
+  }
+
+  public static Map<String, String> getSignAuthHeaders(SignRequest signReq) {
+    Map<String, String> signHeaders = new HashMap<>();
+    authHeaderProviders.forEach(provider -> signHeaders.putAll(provider.getSignAuthHeaders(signReq)));
+    return signHeaders;
+  }
+
   public static void addDefaultHeaders(HttpClientRequest request) {
     request.headers().addAll(getDefaultHeaders());
   }
 
-  public static MultiMap getDefaultHeaders() {
+  public static Map<String, String> getDefaultHeaders() {
     // add token header
-    return new CaseInsensitiveHeaders().add(HEADER_CONTENT_TYPE, "application/json")
-        .add(HEADER_USER_AGENT, "cse-serviceregistry-client/1.0.0")
-        .add(HEADER_TENANT_NAME, ServiceRegistryConfig.INSTANCE.getTenantName())
-        .addAll(authHeaders());
+    Map<String, String> headers = new HashMap<String, String>();
+    headers.put(HEADER_CONTENT_TYPE, "application/json");
+    headers.put(HEADER_USER_AGENT, "cse-serviceregistry-client/1.0.0");
+    headers.put(HEADER_TENANT_NAME, ServiceRegistryConfig.INSTANCE.getTenantName());
+    headers.putAll(authHeaders());
```

Comment:
I'd prefer users need only to implement getSignAuthHeaders. So for deafult getSignAuthHeaders may like this:
{
    if(enableSignHeaders) sign this header
    else call authHeaders method. 
}

If call getSignAuthHeaders  && authHeaders both in sc client, these headers will mixed up and have redundant information.